src/caches/beliefpropagationcache.jl

```diff
@@ -38,12 +37,16 @@ function default_cache_construction_kwargs(alg::Algorithm"bp", ψ::AbstractITens
   return (; partitioned_vertices=default_partitioned_vertices(ψ))
 end
 
-function message_diff(
-  message_a::Vector{ITensor}, message_b::Vector{ITensor}; message_norm=default_message_norm
-)
+function message_diff(message_a::Vector{ITensor}, message_b::Vector{ITensor})
   lhs, rhs = contract(message_a), contract(message_b)
-  norm_lhs, norm_rhs = message_norm(lhs), message_norm(rhs)
-  return 0.5 * norm((denseblocks(lhs) / norm_lhs) - (denseblocks(rhs) / norm_rhs))
+  @assert issetequal(inds(lhs), inds(rhs))
+  c = combiner(inds(lhs))
+  lhs *= c
+  rhs *= c
```

Comment:
The combiner seems unnecessary, `norm` and `dot` should be defined for ITensors already. Also checking the indices seems unnecessary, since `dot(a::ITensor, b::ITensor)` is defined as `(dag(a) * b)[]` so will check the result is a scalar, which can only happy if `a` and `b` share indices.

Reply:
Oh that's neat, I will remove those lines